Google/google_module/google_reports.py

```diff
@@ -62,49 +62,56 @@ class GoogleReports(GoogleTrigger):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.context = PersistentJSON("context.json", self._data_path)
-        self.from_date = self.most_recent_date_seen
         self.service_account_path = self.CREDENTIALS_PATH
         self.scopes = [
             "https://www.googleapis.com/auth/admin.reports.audit.readonly",
             "https://www.googleapis.com/auth/admin.reports.usage.readonly",
         ]
         self.events_sum = 0
+        self.from_date = ""
 
     @property
     def most_recent_date_seen(self):
         now = datetime.now(timezone.utc)
 
         with self.context as cache:
-            most_recent_date_seen_str = cache.get("most_recent_date_seen")
+            app_key_name_in_cache = "most_recent_date_seen_" + self.configuration.application_name.value
+            most_recent_date_seen_str = cache.get(app_key_name_in_cache)
```

Comment:
For a smooth transition, can you add that, if `most_recent_date_seen_<application_name>` is undefined to get `most_recent_date_seen`?